terraform/mssql/provision/main.tf

```diff
@@ -55,11 +55,31 @@ resource "aws_db_instance" "db_instance" {
   storage_type           = var.storage_type
   iops                   = contains(local.valid_storage_types_for_iops, var.storage_type) ? var.iops : null
 
+  parameter_group_name = aws_db_parameter_group.db_parameter_group.name
+
   lifecycle {
     prevent_destroy = true
   }
 
   timeouts {
     create = "60m"
   }
+}
+
+resource "aws_db_parameter_group" "db_parameter_group" {
+  family = local.family
+  # The name cannot be repeated. We need `name_prefix` when upgrading major version.
+  # See `DBParameterGroupAlreadyExists` error:
+  # https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBParameterGroup.html
+  name_prefix = format("rds-mssql-%s", var.instance_name)
```

Comment:
**note:** When prefixing or combining properties to form a new field you can check if their lengths, and accepted characters are compatible.

In this case:
- `--db-instance-identifier` : https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-instance.html#options  
> Must contain from 1 to 63 letters, numbers, or hyphens.
> First character must be a letter.
> Can't end with a hyphen or contain two consecutive hyphens.

- `--db-parameter-group-name` : https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-parameter-group.html#options  
> Must be 1 to 255 letters, numbers, or hyphens.
> First character must be a letter
> Can't end with a hyphen or contain two consecutive hyphens

So they are perfectly compatible and there is no single point of failure. :)

For an example of some fields might present some difficulties when combined see:
https://github.com/cloudfoundry/csb-brokerpak-aws/pull/1042/files#diff-a0ca5d17d658b7dda75c6b790120bf882776543a69995cf8fad541b463d0c527R95-R96

Reply:
Great point! Thank you very much!